mollie-payments-for-woocommerce/includes/mollie/wc/plugin.php

```diff
@@ -33,6 +37,97 @@ class Mollie_WC_Plugin
+            wp_schedule_event( $time, 'daily', 'pending_payment_confirmation_check' );
+        }
+
+        add_action( 'pending_payment_confirmation_check', array(__CLASS__, 'checkPendingPaymentOrdersExpiration'));
+
+
+    }
+
+    /**
+     *
+     */
+    public static function initDb()
+    {
+        global $wpdb;
+        $wpdb->mollie_pending_payment = $wpdb->prefix . self::PENDING_PAYMENT_DB_TABLE_NAME;
+        if(get_option(self::DB_VERSION_PARAM_NAME, '') != self::DB_VERSION){
+
+            global $wpdb;
+            $pendingPaymentConfirmTable = $wpdb->prefix . self::PENDING_PAYMENT_DB_TABLE_NAME;
+            require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+            if($wpdb->get_var("show tables like '$pendingPaymentConfirmTable'") != $pendingPaymentConfirmTable) {
+                $sql = "CREATE TABLE " . $pendingPaymentConfirmTable . " (
+                    `id` int(11) NOT NULL AUTO_INCREMENT,
+                    `post_id` bigint NOT NULL,
+                    `expired_time` int NOT NULL,
+                    UNIQUE KEY id (id)
+                );";
+                dbDelta($sql);
+            }
+            update_option(self::DB_VERSION_PARAM_NAME, self::DB_VERSION);
+        }
+
+
+    }
+
+    /**
+     *
+     */
+    public static function checkPendingPaymentOrdersExpiration()
+    {
+        global $wpdb;
+        $currentDate = new DateTime();
+        $items = $wpdb->get_results("SELECT * FROM {$wpdb->mollie_pending_payment} WHERE expired_time < {$currentDate->getTimestamp()};");
+        foreach ($items as $item){
+            $order =  wc_get_order( $item->post_id );
+            if ($order->get_status() == Mollie_WC_Gateway_Abstract::STATUS_COMPLETED){
+
+                $new_order_status = Mollie_WC_Gateway_Abstract::STATUS_FAILED;
+                $paymentMethodId = get_post_meta( $order->id, '_payment_method_title', true );
+                $molliePaymentId    = get_post_meta( $order->id, '_mollie_payment_id', true );
+
+                $order->add_order_note(sprintf(
+                /* translators: Placeholder 1: payment method title, placeholder 2: payment ID */
+                    __('%s payment failed (%s).', 'mollie-payments-for-woocommerce'),
+                    $paymentMethodId,$molliePaymentId
+                ));
+
+                $order->update_status($new_order_status, '');
+
+                if (get_post_meta($order->id, '_order_stock_reduced', $single = true)) {
+                    // Restore order stock
+                    Mollie_WC_Plugin::getDataHelper()->restoreOrderStock($order);
+
+                    Mollie_WC_Plugin::debug(__METHOD__ . " Stock for order {$order->id} restored.");
+                }
+
+                $wpdb->delete(
+                    $wpdb->mollie_pending_payment,
+                    array(
+                        'post_id' => $order->id,
+                    )
+                );
+            }
+        }
+
+        //reinit job
```

Comment:
Deze was nodig omdat de cronjob tijd via de settings aan te passen was. Nu deze setting weg is, hebben we deze reinit ook weggehaald.